azure/services/scalesetvms/scalesetvms.go

```diff
@@ -143,3 +257,32 @@ func (s *Service) Delete(ctx context.Context) error {
 	s.Scope.DeleteLongRunningOperationState(instanceID, serviceName, infrav1.DeleteFuture)
 	return nil
 }
+
+// VMSSFlexVMGetter is an interface for getting all the required information to create/update/delete an Azure resource.
+type VMSSFlexVMGetter struct {
+	Name          string
+	ResourceGroup string
+}
+
+// ResourceName returns the name of the resource.
+func (vm *VMSSFlexVMGetter) ResourceName() string {
+	return vm.Name
+}
+
+// OwnerResourceName returns the name of the resource that owns the resource
+// in the case that the resource is an Azure subresource.
+func (vm *VMSSFlexVMGetter) OwnerResourceName() string {
+	return ""
+}
+
+// ResourceGroupName returns the name of the resource group the resource is in.
+func (vm *VMSSFlexVMGetter) ResourceGroupName() string {
+	return vm.ResourceGroup
+}
+
+// Parameters takes the existing resource and returns the desired parameters of the resource.
+// If the resource does not exist, or we do not care about existing parameters to update the resource, existing should be nil.
+// If no update is needed on the resource, Parameters should return nil.
+func (vm *VMSSFlexVMGetter) Parameters(ctx context.Context, existing interface{}) (params interface{}, err error) {
```

Comment:
can you add a comment here on why Parameters is a no-op? Assuming it's because #2720 hasn't been implemented yet